src/app/model/migrations/migrations/to0002/MigrationToVersion0002.ts

```diff
@@ -0,0 +1,164 @@
+
+export class MigrationToVersion0002 implements IMigration {
+  private evaluateVideoPhotoItemSteps(
+    activityItems: Array<PipelineItem>,
+  ): Array<string> {
+    const mediaItemSteps = activityItems.reduce<Array<string>>(
+      (result, item, step) => {
+        if (isVideoOrPhotoItem(item)) {
+          result.push(step.toString());
+        }
+
+        return result;
+      },
+      [],
+    );
+
+    return mediaItemSteps;
+  }
+
+  private evaluateAudioItemSteps(
+    activityItems: Array<PipelineItem>,
+  ): Array<string> {
+    const mediaItemSteps = activityItems.reduce<Array<string>>(
+      (result, item, step) => {
+        if (isAudioItem(item)) {
+          result.push(step.toString());
+        }
+
+        return result;
+      },
+      [],
+    );
+
+    return mediaItemSteps;
+  }
+
+  private async processVideoPhotoItems(activityFrom: ActivityState) {
+    const videoPhotoItemSteps = this.evaluateVideoPhotoItemSteps(
+      activityFrom.items,
+    );
+
+    for await (const step of videoPhotoItemSteps) {
+      const mediaFile = activityFrom.answers[step]?.answer;
+      const hasAnswer = !!mediaFile?.fileName;
+
+      if (!hasAnswer) {
+        continue;
+      }
+
+      if (mediaFile.uri) {
+        if (IS_IOS) {
+          /**
+           * For iOS, it is required to moved files from /tmp/ folder (created by react-native-image-picker).
+           * For Android, files' fileName property be renamed because react-native-image-picker assigns them wrong names.
+           */
+          await moveMediaFileToCache(mediaFile.fileName, mediaFile.uri);
+        } else {
+          renameMediaFile(mediaFile);
+        }
+      }
+    }
+  }
+
+  private processAudioFiles(activityFrom: ActivityState) {
+    const audioItemSteps = this.evaluateAudioItemSteps(activityFrom.items);
+
+    for (const step of audioItemSteps) {
+      const audioFile = activityFrom.answers[step]?.answer;
+      const hasAnswer = !!audioFile?.fileName;
+
+      if (!hasAnswer) {
+        continue;
+      }
+
+      renameMediaFile(audioFile);
+    }
+  }
+
+  private processUserActions(userActions: Array<UserAction>) {
+    for (const action of userActions) {
+      /**
+       * In user actions, we should only change the fileName and there is no need
+       * to move the file because the file might have been already deleted.
+       */
+      const shouldRenamePhotoVideo =
+        IS_ANDROID &&
+        action.type === 'SET_ANSWER' &&
+        isVideoOrPhotoItem(action.payload.answer);
+
+      const shouldRenameAudio =
+        action.type === 'SET_ANSWER' && isAudioItem(action.payload.answer);
+
+      const shouldRenameMediaFile = shouldRenamePhotoVideo || shouldRenameAudio;
+
+      if (shouldRenameMediaFile) {
+        renameMediaFile(action.payload.answer.value.answer!);
+      }
+    }
+  }
+
+  private async getUpdatedActivityState(
+    activityFrom: ActivityState,
+  ): Promise<ActivityState> {
+    await Promise.all([
```

Comment:
Maybe it worth to cover every internal function with try-carch to allow to Promise.all to complete their jobs?

And it's defenitelly worth to add more logs in every function about:  pathFrom, pathTo, renameFileNameFrom, renameFileNameTo, "move (un)succesfull" "rename un(succesfull)", an so on, in order to get possibility to analyze logs in case of issues.

Reply:
I don't think it is worth it, at least for `processUserActions` because if `processVideoPhotoItems`/`processAudioFiles` fails there is no need to even call `processUserActions` because it basically does the same job `processAudioFiles`/`processAudioFiles` does but in the user actions